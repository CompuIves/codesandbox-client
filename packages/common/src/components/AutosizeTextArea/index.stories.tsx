import * as React from 'react';
import { storiesOf } from '@storybook/react';
import AutosizeTextArea from './';
import { ThemeDecorator } from '../../stories/decorators';

const stories = storiesOf('components/Input', module).addDecorator(
  ThemeDecorator
);

stories.add('Basic AutosizeTextArea', () => <AutosizeTextArea />);
