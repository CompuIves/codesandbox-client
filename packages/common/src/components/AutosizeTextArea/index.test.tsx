import 'jest-styled-components';
import React from 'react';
import mountWithTheme from '../../test/themeMount';
import AutosizeTextArea from './';

let wrapper;

beforeEach(() => {
  wrapper = mountWithTheme(<AutosizeTextArea />);
});

describe('<AutosizeTextArea /> rendering', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
