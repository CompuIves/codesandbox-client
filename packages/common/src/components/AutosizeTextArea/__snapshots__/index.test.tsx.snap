// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutosizeTextArea /> rendering renders correctly 1`] = `
.c0 {
  -webkit-transition: 0.3s ease border-color;
  transition: 0.3s ease border-color;
  background-color: rgba(0,0,0,0.3);
  color: white;
  border: none;
  outline: none;
  border-radius: 4px;
  padding: 0.25em;
  width: inherit;
  box-sizing: border-box;
  border: 1px solid rgba(0,0,0,0.1);
}

.c0:focus {
  border-color: rgba(64,169,243,0.4);
}

<Styled(TextareaAutosize)>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  ",
            "
  transition: 0.3s ease border-color;
  background-color: ",
            [Function],
            ";
  color: ",
            [Function],
            ";
  border: none;
  outline: none;
  border-radius: 4px;
  padding: 0.25em;
  width: ",
            [Function],
            ";
  box-sizing: border-box;

  border: 1px solid
    ",
            [Function],
            ";

  &:focus {
    border-color: ",
            [Function],
            ";
  }
",
            ";
",
          ],
        },
        "displayName": "Styled(TextareaAutosize)",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <TextareaAutosize
      className="c0"
      inputRef={[Function]}
      onChange={[Function]}
      onHeightChange={[Function]}
      useCacheForDOMMeasurements={false}
    >
      <textarea
        className="c0"
        onChange={[Function]}
        style={
          Object {
            "height": 2,
          }
        }
      />
    </TextareaAutosize>
  </StyledComponent>
</Styled(TextareaAutosize)>
`;
